pkg/ebpf/tracer_windows.go

```diff
@@ -115,22 +122,41 @@ func printStats(stats []network.ConnectionStats) {
 
 // GetActiveConnections returns all active connections
 func (t *Tracer) GetActiveConnections(clientID string) (*network.Connections, error) {
-	connStatsActive, connStatsClosed, err := t.driverInterface.GetConnectionStats()
+	t.bufferLock.Lock()
+	defer t.bufferLock.Unlock()
+
+	activeConnStats, closedConnStats, err := t.driverInterface.GetConnectionStats(t.connStatsActive[:0], t.connStatsClosed[:0])
 	if err != nil {
 		log.Errorf("failed to get connnections")
 		return nil, err
 	}
 
-	for _, connStat := range connStatsClosed {
+	for _, connStat := range closedConnStats {
 		t.state.StoreClosedConnection(connStat)
 	}
 
 	// check for expired clients in the state
 	t.state.RemoveExpiredClients(time.Now())
-	conns := t.state.Connections(clientID, uint64(time.Now().Nanosecond()), connStatsActive, t.reverseDNS.GetDNSStats())
+	conns := t.state.Connections(clientID, uint64(time.Now().Nanosecond()), activeConnStats, t.reverseDNS.GetDNSStats())
+	t.checkBufferResize(len(activeConnStats), len(closedConnStats))
```

Comment:
I could be wrong, but I think there is a slice ownership problem here. If `connStatsActive` is properly sized and `DriverInterface.GetConnectionStats()` doesn't need to grow the underlying array, then `connStatsActive` and `activeConnStats` point to the same underlying array. There is at least one case (the new client case) of `State.Connections()` that returns the `[]ConnectionStats` slice untouched, and thus `conns` will also point to the same underlying array.

That array will get clobbered the next time `DriverInterface.GetConnectionStats()` runs. The fix here may be to have `State.Connections()` always copy.

@p-lambert can you check my logic?

Reply:
`State.Connections` should not hold any references to the passed in connections after it finishes. It doesn't exactly make a copy of the connections that come in, but  it doesn't hold on to references to any of those passed in connections.

FWIW we are doing something very similar in the linux tracer.

Reply:
The return value of `State.Connections` (`conns` in this case) is exactly one of the arguments(`activeConnStats`) in the new client case. That return value is returned from this function, but would get clobbered later.

Let me see if I can write a failing test case.

Reply:
OK. I wrote a failing test case: https://gist.github.com/brycekahle/181b25f389c6da07a3af7c2a96e6e7e3

Reply:
Spoke to Lee and we're going to merge this as-is and then I'm going to work on setting up a fix for state.go in a separate PR.